config/runtime.exs

```diff
@@ -0,0 +1,188 @@
+          # dhfile is not supported for tls 1.3
+          # https://www.erlang.org/doc/man/ssl.html#type-dh_file
+          dhfile: Teiserver.ConfigHelpers.get_env("HTTPS_DH_FILE_PATH", "/etc/ssl/dhparam.pem")
+        ],
+    http: [:inet6, port: Teiserver.ConfigHelpers.get_env("PORT", "4000", :int)],
+    secret_key_base: Teiserver.ConfigHelpers.get_env("HTTP_SECRET_KEY_BASE")
+
+  config :teiserver, Teiserver.Account.Guardian,
+    issuer: Teiserver.ConfigHelpers.get_env("GUARDIAN_ISSUER", "teiserver"),
+    secret_key: Teiserver.ConfigHelpers.get_env("GUARDIAN_SECRET_KEY")
+
+  if Teiserver.ConfigHelpers.get_env("ENABLE_EMAIL_INTEGRATION", :bool) do
+    smtp_server = Teiserver.ConfigHelpers.get_env("SMTP_HOSTNAME")
+
+    config :teiserver, Teiserver.Mailer,
+      adapter: Bamboo.SMTPAdapter,
+      noreply_name: "Beyond All Reason",
+      noreply_address: "noreply@#{domain_name}",
+      contact_address: "info@#{domain_name}",
+      server: smtp_server,
+      hostname: smtp_server or Teiserver.ConfigHelpers.get_env("SMTP_SERVER"),
+      # port: 1025,
+      port: Teiserver.ConfigHelpers.get_env("SMTP_PORT", "587", :int),
+      username: Teiserver.ConfigHelpers.get_env("SMTP_USERNAME"),
+      password: Teiserver.ConfigHelpers.get_env("SMTP_PASSWORD"),
+      # tls: :if_available, # can be `:always` or `:never`
+      # can be `:always` or `:never`
+      tls: :always,
+      tls_verify:
+        if(Teiserver.ConfigHelpers.get_env("SMTP_TLS_VERIFY", true, :bool),
+          do: :verify_peer,
+          else: :verify_none
+        ),
+      # or {":system", ALLOWED_TLS_VERSIONS"} w/ comma seprated values (e.g. "tlsv1.1,tlsv1.2")
+      allowed_tls_versions: [:"tlsv1.2"],
+      # can be `true`
+      ssl: false,
+      retries: 1,
+      # can be `true`
+      no_mx_lookups: false,
+      # auth: :if_available # can be `always`. If your smtp relay requires authentication set it to `always`.
+      auth: :always
+  end
+
+  log_root_path = "/var/log/teiserver/"
+
+  config :logger,
+    backends: [
+      {LoggerFileBackend, :error_log},
+      {LoggerFileBackend, :notice_log},
+      {LoggerFileBackend, :info_log},
+      :console
+    ]
+
+  # Do not print debug messages in production
+  config :logger,
+    format: "$date $time [$level] $metadata $message\n",
+    metadata: [:request_id, :user_id],
+    level: :info
+
+  config :logger, :error_log,
+    path: "#{log_root_path}error.log",
+    format: "$date $time [$level] $metadata $message\n",
+    metadata: [:request_id, :user_id],
+    level: :error
+
+  config :logger, :notice_log,
+    path: "#{log_root_path}notice.log",
+    format: "$date $time [$level] $metadata $message\n",
+    metadata: [:request_id, :user_id],
+    level: :notice
+
+  config :logger, :info_log,
+    path: "#{log_root_path}info.log",
+    format: "$date $time [$level] $metadata $message\n",
+    metadata: [:request_id, :user_id],
+    level: :info
+
+  if enable_discord_bridge do
```

Comment:
This seems to miss the `config :teiserver, DiscordBridgeBot,` part I have https://github.com/beyond-all-reason/ansible-teiserver/blob/main/roles/teiserver/templates/prod.secret.exs.j2#L73-L77

I'm not sure how without that it would know to what server to connect, how have you tested it?

Reply:
I can't really test the actual integration since I don't have access to any token. Are there some test credentials somewhere I could use?
The only test I've done is that, when the discord integration is disabled, I can start the server normally and it won't complain that it's missing stuff for discord.

I've added the configuration for the bridge in 3b72d871c2b56101646b33418d7af5cd189c7e3b

Reply:
You can follow https://github.com/beyond-all-reason/ansible-teiserver?tab=readme-ov-file#discord to create discord for testing, there are also instructions there for email testing etc.

Reply:
I've amended the commit for the bot config, followed your steps and now it seems to work.
![2024-05-11-475x166-scrot](https://github.com/beyond-all-reason/teiserver/assets/1531763/96482f79-82e4-4feb-bfef-638f3b36cb93)